guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/main/java/org/guvnor/m2repo/client/widgets/ArtifactListPresenterImpl.java

```diff
@@ -36,58 +35,67 @@
 public class ArtifactListPresenterImpl
         implements ArtifactListPresenter {
 
-    @Inject
-    protected ArtifactListViewImpl view;
+    static final boolean DEFAULT_ORDER_ASCENDING = true;
+    private final ArtifactListView view;
 
-    @Inject
-    private Caller<M2RepoService> m2RepoService;
+    private final Caller<M2RepoService> m2RepoService;
 
-    @Inject
-    private Event<NotificationEvent> notification;
+    private final Event<NotificationEvent> notification;
 
+    // Initialized lazily on the first refresh
     private RefreshableAsyncDataProvider dataProvider;
 
+    @Inject
+    public ArtifactListPresenterImpl( ArtifactListView view, Caller<M2RepoService> m2RepoService, Event<NotificationEvent> notification ) {
+        this.view = view;
+        this.m2RepoService = m2RepoService;
+        this.notification = notification;
+    }
+
     @PostConstruct
     public void init() {
         view.init( this );
+        dataProvider = new RefreshableAsyncDataProvider( view, m2RepoService );
+        dataProvider.addDataDisplay( view.getDisplay() );
+    }
+
+    @Override
+    public ArtifactListView getView() {
+        return view;
     }
 
     @Override
     public void refresh() {
-        // We couldn't attach the Display to the DataProvider in the @PostConstruct method
-        // as this initializes the Display with data using AsyncDataProvider.onRangeChanged()
-        // at which time the View may not have been initialised fully.
```

Comment:
@yurloc hehe. as if I knew ;)... I recall it has something to do with the re-use of the ```ArtifactListPresenter``` and related ```ArtifactListView``` in the Project Editor's "Add dependency from Repository" view.. see [here](https://github.com/droolsjbpm/kie-wb-common/blob/master/kie-wb-common-screens/kie-wb-common-project-editor/kie-wb-common-project-editor-client/src/main/java/org/kie/workbench/common/screens/projecteditor/client/forms/DependencyListWidget.java#L77).